src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/HalfConverter.cs

```diff
@@ -0,0 +1,228 @@
+                    {
+                        return value;
+                    }
+
+                    return ReadCore(ref reader);
+                }
+                else if ((JsonNumberHandling.AllowNamedFloatingPointLiterals & handling) != 0)
+                {
+                    if (!TryGetFloatingPointConstant(ref reader, out Half value))
+                    {
+                        ThrowHelper.ThrowFormatException(NumericType.Half);
+                    }
+
+                    return value;
+                }
+            }
+
+            return Read(ref reader, Type, options);
+        }
+
+        internal override void WriteNumberWithCustomHandling(Utf8JsonWriter writer, Half value, JsonNumberHandling handling)
+        {
+            if ((JsonNumberHandling.WriteAsString & handling) != 0)
+            {
+#if NET8_0_OR_GREATER
+                const byte Quote = JsonConstants.Quote;
+                Span<byte> buffer = stackalloc byte[MaxFormatLength + 2];
+#else
+                const char Quote = (char)JsonConstants.Quote;
+                Span<char> buffer = stackalloc char[MaxFormatLength + 2];
+#endif
+                buffer[0] = Quote;
+                Format(buffer.Slice(1), value, out int written);
+
+                int length = written + 2;
+                buffer[length - 1] = Quote;
+                writer.WriteRawValue(buffer.Slice(0, length));
+            }
+            else if ((JsonNumberHandling.AllowNamedFloatingPointLiterals & handling) != 0)
+            {
+                WriteFloatingPointConstant(writer, value);
+            }
+            else
+            {
+                WriteCore(writer, value);
+            }
+        }
+
+        private static bool TryGetFloatingPointConstant(ref Utf8JsonReader reader, out Half value)
+        {
+            Span<byte> buffer = stackalloc byte[MaxFormatLength];
+            int written = reader.CopyValue(buffer);
+
+            return JsonReaderHelper.TryGetFloatingPointConstant(buffer.Slice(0, written), out value);
+        }
+
+        private static void WriteFloatingPointConstant(Utf8JsonWriter writer, Half value)
+        {
+            if (Half.IsNaN(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.NaNValue);
+            }
+            else if (Half.IsPositiveInfinity(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.PositiveInfinityValue);
+            }
+            else if (Half.IsNegativeInfinity(value))
+            {
+                writer.WriteNumberValueAsStringUnescaped(JsonConstants.NegativeInfinityValue);
+            }
+            else
+            {
+                WriteCore(writer, value);
+            }
+        }
+
+        // Half.TryFormat/TryParse(ROS<byte>) are not available on .NET 7
+        // we need to use Half.TryFormat/TryParse(ROS<char>) in that case.
+        private static bool TryParse(
+#if NET8_0_OR_GREATER
```

Comment:
Clarifying a bit more what I meant in https://github.com/dotnet/runtime/pull/88962/files#r1266760827, I think `TryParse` could be made to only accept a `ROS<byte>` argument and only transcode inside the method if `!NET8_0_OR_GREATER`.